tests/unit/test-class-wc-payment-gateway-wcpay.php

```diff
@@ -706,6 +731,77 @@ function ( $order ) {
 		$this->assertFalse( $afterpay_method->is_enabled_at_checkout( 'US' ) );
 	}
 
+	public function test_payment_methods_enabled_based_on_currency_limits() {
+		WC_Helper_Site_Currency::$mock_site_currency = 'USD';
+
+		WC()->session->init();
+		WC()->cart->empty_cart();
+		// Total is 100 USD, which is above both payment methods (Affirm and AfterPay) minimums.
+		WC()->cart->add_to_cart( WC_Helper_Product::create_simple_product()->get_id(), 10 );
```

Comment:
The amount could be defined a bit more "dynamically" (in case the limits change in the future) but I didn't want to overcomplicate things, lmk what do you think.